core/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -470,11 +469,13 @@ protected final void disconnectFromNodeChannel(final Channel channel, final Exce
     }
 
     protected Channel nodeChannel(DiscoveryNode node, TransportRequestOptions options) throws ConnectTransportException {
-        NodeChannels nodeChannels = connectedNodes.get(node);
-        if (nodeChannels == null) {
-            throw new NodeNotConnectedException(node, "Node not connected");
+        try (Releasable ignored = connectionLock.acquire(node.getId())) {
```

Comment:
That's correct. But I was worried between the HB-edge between a new request incoming (-> `#nodeChannel()') and a concurrent disconnect (`#disconnectFromNode`).